openedx/core/djangoapps/theming/helpers.py

```diff
@@ -298,7 +319,16 @@ def is_comprehensive_theming_enabled():
     Returns:
          (bool): True if comprehensive theming is enabled else False
     """
+    # Import is placed here to avoid circular import
+    from openedx.core.djangoapps.theming.microsite_transition_helpers import current_request_has_associated_site_theme
+
+    # We need to give priority to theming over microsites
+    if current_request_has_associated_site_theme():
+        return True
+
     # Disable theming for microsites
+    # We need to check microsite requests here becasue of we do not then
+    # DEFAULT_SITE_THEME will take over microsite
     if microsite.is_request_in_microsite():
         return False
```

Comment:
@douglashall we need this block because there are these scenarios

- Theming is enabled and current site has associated `SiteTheme` and `microsite`, this case is handled by [this](https://github.com/edx/edx-platform/blob/saleem-latif/MAYN-207-transition-to-theming/openedx/core/djangoapps/theming/helpers.py#L319-L320) block.
```
    if settings.ENABLE_COMPREHENSIVE_THEMING and current_request_has_associated_site_theme():
        return True
```
- Theming is enabled and current site does not have any associated `SiteTheme` but it does have a `microsite`, this case is handled by [this](https://github.com/edx/edx-platform/blob/saleem-latif/MAYN-207-transition-to-theming/openedx/core/djangoapps/theming/helpers.py#L322-L325) block.
```
    if microsite.is_request_in_microsite():
        return False
```
- Then finally we have the final case, Current site has neither associate `SiteTheme` nor `microsite`, So, this case (to whether we need to apply `DEFAULT_SITE_THEME`) is handled by [this](https://github.com/edx/edx-platform/blob/saleem-latif/MAYN-207-transition-to-theming/openedx/core/djangoapps/theming/helpers.py#L327) return statement.
```
return settings.ENABLE_COMPREHENSIVE_THEMING
```